### DISCUSSION TITLE
Reset loop after installing vselect module [Solved]

### DISCUSSION THREAD

**Discussion by coopercummings6** [sanni/cartreader#733] [Hardware]
Today I recieved pcbs for the vselect module in the mail, then assembled one and installed it in my previously working cart reader V5. After installing the vselect module my cart reader powers on, however the screen flashes off and on once every 1-2 seconds, and the software resets as if I had hit the reset button, the menu goes back to the top item at the highest level even if I had selected something else. When I check VCC with a multimeter it toggles between 0V and ~3.3V when it does this. I have attached pictures of my install. Does anyone have any suggestions on how to troubleshoot this? The only thing I can think of is that I may have managed to install the chip backwards, I could only see one dot on the chip and lined it up with the dot on the board, but the dot is much larger than pin 1 indicators dots I have seen on other parts, usually when I see one this size it is opposite the actual pin 1 indicator dot, usually an artifact from injection molding the plastic housing or something.

NOTE: While doing one final sanity check and beeping everything out with my multimeter I found a solution to my problem, but I already had this typed up so I will post it anyways in case it is helpful to anyone else who encounters similar issues.

The problem was a bad ground, when installing the board I incorrectly assumed that the solder points to mechanically secure the housing of the power switch would be connected to the ground plane, as it is on many devices I have previously worked on. When I double checked continuity with my multimeter I noticed that this is not the case with the current version of the open source cart reader, after moving my JP2 wire to an actual ground point the reset loop issue has been resolved.

I would attach pictures, but the file upload tool seems to be acting up.

**Comment by Ancyker** [comment]
Yeah, that issue actually tripped me up when designing the PCB. I was going to use them as the ground, requiring only the digital pin jumper, but they aren't grounded. I think it's a good thing they aren't, though. I messed around with those switches, and it was easy to short them to the housing by heating them.

I've been thinking about reverting my change to vselect which has it on the top of the board. My original design had an error: I flipped the 5V and 3.3V inputs. But the wiring is actually a lot easier on the bottom of the board:

![2022-12-20 13 02 07](https://user-images.githubusercontent.com/7105333/224365930-626bac34-0e6c-4ed0-9043-b2c61e5cff67.jpg)
